File: src/test/java/dev/retrotv/random/generator/PINGeneratorTest.java
package dev.retrotv.random.generator;

import org.junit.jupiter.api.DisplayName;
import org.junit.jupiter.api.RepeatedTest;

import java.security.SecureRandom;

import static org.junit.jupiter.api.Assertions.*;

class PINGeneratorTest {

    @RepeatedTest(value = 100, name = "{displayName}, {currentRepetition}/{totalRepetitions}")
    @DisplayName("PIN 생성 테스트")
    void test_pinGenerator() {
        PINGenerator pinGenerator = new PINGenerator(new SecureRandom());
        String pin = pinGenerator.generate();

        assertEquals(4, pin.length());
        assertFalse(pin.matches(".*[A-Z]+"));
        assertFalse(pin.matches(".*[a-z]+"));
        assertTrue(pin.matches(".*[0-9]+"));
        assertFalse(pin.matches(".*[^A-Za-z0-9]+"));
    }
}

File: src/test/kotlin/dev/retrotv/random/generator/PINGeneratorTest.kt
package dev.retrotv.random.generator

import java.security.SecureRandom
import org.junit.jupiter.api.*
import kotlin.test.*

class PINGeneratorTest {

    @RepeatedTest(value = 100, name = "{displayName}, {currentRepetition}/{totalRepetitions}")
    @DisplayName("PIN 생성 테스트")
    fun test_generate() {
        val pinGenerator = PINGenerator(SecureRandom())
        pinGenerator.setLength(6)
        val pin = pinGenerator.generate()

        assertTrue(pin.length == 6)
        assertFalse(pin.contains(Regex(".*[A-Z]+")))
        assertFalse(pin.contains(Regex(".*[a-z]+")))
        assertTrue(pin.contains(Regex(".*[0-9]+")))
        assertFalse(pin.contains(Regex(".*[^A-Za-z0-9]+")))
    }
}